zoltar-tensorflow/src/main/java/com/spotify/zoltar/tf/SimpleTensorFlowPredictorBuilder.java

```diff
@@ -0,0 +1,134 @@
+import com.spotify.futures.CompletableFutures;
+import com.spotify.zoltar.FeatureExtractFns;
+import com.spotify.zoltar.ModelLoader;
+import com.spotify.zoltar.Prediction;
+import com.spotify.zoltar.Predictor;
+import com.spotify.zoltar.featran.FeatranExtractFns;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import org.tensorflow.Session;
+import org.tensorflow.Tensor;
+import org.tensorflow.Tensors;
+import org.tensorflow.example.Example;
+
+/**
+ * Helps to build a simple TensorFlow predictor.
+ * In this context, a simple TensorFlow predictor has the following characteristics:
+ * - Has a single input op
+ * - Has a single output op
+ * - Uses Featran to extract and consume Example objects
+ * The output of the Predictor will always be a {@link JTensor}.
+ */
+public class SimpleTensorFlowPredictorBuilder<InputT> {
+
+  private FeatureExtractFns.ExtractFn<InputT, Example> extractFn;
+  private String inputOp;
+  private String predictOp;
+  private ModelLoader<TensorFlowModel> modelLoader;
+
+  /**
+   * Parameters for FeatranExtractFns.
+   *
+   * @param featureSpec Featran's {@link FeatureSpec}.
+   * @param settings    JSON settings from a previous session.
+   */
+  public SimpleTensorFlowPredictorBuilder<InputT> withFeatranExtractFns(
+      final FeatureSpec<InputT> featureSpec,
+      final String settings) {
+    extractFn = FeatranExtractFns.example(featureSpec, settings);
+    return this;
+  }
+
+  /**
+   * Name of the input op node in the TF graph.
+   */
+  public SimpleTensorFlowPredictorBuilder<InputT> withInputOp(final String inputOp) {
+    this.inputOp = inputOp;
+    return this;
+  }
+
+  /**
+   * Name of the predict op node in the TF graph.
+   */
+  public SimpleTensorFlowPredictorBuilder<InputT> withPredictOp(final String predictOp) {
+    this.predictOp = predictOp;
+    return this;
+  }
+
+  /**
+   * Loads a saved TensorFlow {@link org.tensorflow.SavedModelBundle}, can be a URI to a local
+   * filesystem, resource, GCS etc.
+   */
+  public SimpleTensorFlowPredictorBuilder<InputT> withModelUri(final String modelUri) {
+    modelLoader = TensorFlowLoader.create(modelUri);
+    return this;
+  }
+
+  private static JTensor predictFn(final TensorFlowModel model,
+                                   final Example example,
+                                   final String inputOp,
+                                   final String predictOp) {
+    final byte[][] b = new byte[1][];
+    b[0] = example.toByteArray();
+    try (final Tensor<String> t = Tensors.create(b)) {
+      final Session.Runner runner = model.instance().session().runner()
+          .feed(inputOp, t);
+      final Map<String, JTensor> result = TensorFlowExtras.runAndExtract(runner, predictOp);
+      return result.get(predictOp);
+    }
+  }
+
+  /**
+   * Builds the Predictor.
+   */
+  public Predictor<InputT, JTensor> build() {
```

Comment:
I don't necessarily like this API, this looks like a builder, but at the same time always requires all the setters to be called - otherwise will fail at build runtime with null check, so even tho the API allows to build "partially" filled objects, those are invalid, strikes me as something we should avoid?